torchgeo/datasets/nccm.py

```diff
@@ -0,0 +1,244 @@
+"""Northeastern China Crop Map Dataset."""
+
+import glob
+import os
+from collections.abc import Iterable
+from typing import Any, Callable, Optional, Union
+
+import matplotlib.pyplot as plt
+import torch
+from matplotlib.figure import Figure
+from rasterio.crs import CRS
+
+from .geo import RasterDataset
+from .utils import BoundingBox, download_url, extract_archive
+
+
+class NCCM(RasterDataset):
+    """The Northeastern China Crop Map Dataset.
+
+    Link: https://www.nature.com/articles/s41597-021-00827-9
+
+    This dataset produced annual 10-m crop maps of the
+    major crops (maize, soybean, and rice)
+    in Northeast China from 2017 to 2019, using hierarchial mapping strategies,
+    random forest classifiers, interpolated and
+    smoothed 10-day Sentinel-2 time series data and
+    optimized features from spectral, temporal and
+    textural characteristics of the land surface.
+    The resultant maps have high overall accuracies (OA)
+    based on ground truth data. The dataset contains information
+    specific to the three years: 2017,2018,2019.
+
+    The dataset contains 4 classes:
+    0: paddy rice
+    1: maize
+    2: soybean
+    3: others
+
+
+    Dataset format:
+    1) Three .TIF files containing the labels
+    2) JavaScript code to download images from the dataset.
+
+    If you use this dataset in your research, please use the corresponding citation:
+    * You, N., Dong, J., Huang, J. et al. The 10-m crop type maps in
+    Northeast China during 2017-2019.
+    Sci Data 8, 41 (2021). https://doi.org/10.1038/s41597-021-00827-9
+    """
+
+    filename_regex = r"CDL(?P<year>\d{4})_clip"
+    filename_glob = "CDL*.tif"
+    # there is only a single zipfolder, none for the files
+    zipfile_glob = "13090442.zip"
+
+    date_format = "%Y"
+    is_image = False
+
+    # need to add url
+    url = "https://figshare.com/ndownloader/articles/13090442/versions/1"
+
+    # md5s = {
+    #     2017: "a85341fa6248fd7e0badab6c",
+    #     2018: "478d10786aa798fb11693ec1",
+    #     2019: "42e483fdc8239d22dba7020f"
+    # }
+
+    md5s = {"main": "eae952f1b346d7e649d027e8139a76f5"}
+
+    years = [2017, 2018, 2019]
+
+    cmap = {
+        0: (0, 0, 0, 0),
+        1: (0, 255, 0, 255),
+        2: (255, 0, 0, 255),
+        3: (255, 255, 0, 255),
+        4: (255, 0, 255, 255),
+    }
+
+    def __init__(
+        self,
+        paths: Union[str, Iterable[str]] = "data",
+        crs: Optional[CRS] = None,
+        res: Optional[float] = None,
+        years: list[int] = [2019],
```

Comment:
Wondering if this parameter is actually useful. You have to download all years anyway. And the sampler can be told which years you care about.

Reply:
Hi Adam, we kept years as a parameter to prevent invalid years. We have only 3 years: 2017,18,19. we have a test case where we enter 1996, and that should raise an exception. is that fine?

Reply:
But if `years` is no longer a parameter, then we no longer need to prevent invalid years...

Reply:
so I can remove the test case altogether?